shtdlib.sh

```diff
@@ -178,6 +179,56 @@ function umask_decorator {
     return 1
 }
 
+# Bash behaviour option decorator
+# Allows changing/setting bash options for a function code block restoring
+# the original once execution of the function and it's calls is complete.
+# Requires an option name (see shopt) and a truthyness vale "true"/"false" or
+# other command that returns 0/1.
+# variables shopt_decorator_option_name and shopt_decorator_option_value
+# To use this add a line like the following (without #) as the first line of a function
+# shopt_decorator "${FUNCNAME[0]}" "${@:-}" && return
+function shopt_decorator {
+    if [ -n "${shopt_decorator_option_value:-}" ]  && [ -n "$(shopt -o "${shopt_decorator_option_name:-}")" ] ; then
+        if [ "${FUNCNAME[0]}" != "${FUNCNAME[2]}" ] ; then
+            if shopt -qo "${shopt_decorator_option_name}" ; then
```

Comment:
I completely missed the `-q` flag - that would have simplified my code so much!

Reply:
I know, I saw that and thought the same.